launcher.py

```diff
@@ -0,0 +1,405 @@
+        name (str): The name of the dependency.
+        package_name (dict): { platform: package_name }.
+        check_cmd (dict): { platform: list of command or function to check if the dependency is installed }.
+        install_cmd (dict): { platform: list of command or function to install the dependency }.
+
+        if package_name is not provided, it will be set to the name.
+
+        if check_cmd is not provided
+            package_name --version will be used to check if the dependency is installed
+
+        if install cmd is not provided
+            for debian based: apt-get
+            for fedora/rhel: dnf
+            for macos: brew
+            for windows: msys2
+    """
+
+    def __init__(self, name, package_name=None, check_cmd=None, install_cmd=None):
+        self.name = name
+        self.package_name = package_name or {}
+        self.check_cmd = check_cmd or {}
+        self.install_cmd = install_cmd or {}
+        self._installers = {
+            "Linux": self._get_install_commands_linux,
+            "Darwin": self._get_install_commands_darwin,
+            "Windows": self._get_install_commands_windows,
+        }
+        # Mapping of Linux distributions to package managers
+        self._linux_distro_map = {
+            "ubuntu": "sudo apt-get install -y",
+            "debian": "sudo apt-get install -y",
+            "kali": "sudo apt-get install -y ",
+            "pop": "sudo apt-get install -y ",
+            "elementary": "sudo apt-get install -y ",
+            "mint": "sudo apt-get install -y",
+            "fedora": "sudo dnf install -y",
+            "rhel": "sudo dnf install -y",
+            "centos": "sudo dnf install -y",
+            "rocky": "sudo dnf install -y",
+            "alma": "sudo dnf install -y",
+            "arch": "sudo pacman -S --needed --noconfirm",
+            "manjaro": "sudo pacman -S --needed --noconfirm",
+            "endeavouros": "sudo pacman -S --needed --noconfirm",
+            "garuda": "sudo pacman -S --needed --noconfirm",
+            "opensuse": "sudo zypper install -y",
+            "suse": "sudo zypper install -y",
+            "alpine": "sudo apk add",
+            "solus": "sudo eopkg install -y",
+            "void": "sudo xbps-install -y",
+            "clearlinux": "sudo swupd bundle-add",
+        }
+
+    def check_dependency(self, system, install=False):
+        """
+        Checks if the dependency is installed. If not, installs it if `install` is True.
+        """
+        commands = self._get_check_commands(system)
+        try:
+            for cmd in commands:
+                if type(cmd) == str:
+                    run_command(cmd)
+                else:
+                    cmd()
+
+        except (FileNotFoundError, subprocess.CalledProcessError):
+            if not install:
+                print(
+                    f"{self.name} is not installed. Use --install to install dependencies automatically or install manually."
+                )
```

Comment:
This should be an error level log. Could you use a logger instead of printing? `loguru` is usually pretty nice.

Reply:
I think we should use logging library of the python.
`loguru` will need installation.